src/toolbar/Toolbar.jsx

```diff
@@ -1,15 +1,254 @@
+            return (
+                <div ref="chamelToolbar" className="chamel-toolbar" style={{position: 'absolute'}}>
+                    <ToolbarGroup>
+                        {displayIcons}
+                    </ToolbarGroup>
+                </div>
+            );
+        } else if (this.state.totalIconWidth > this.state.mainContainerWidth) {
+
+            let displayArrowLeft = null,
+                displayArrowRight = null,
+                totalDisplayIconWidth = 0,
+                idx = this.state.startIconIndex;
+
+            // Let's use the maxIconsDisplay as our limit on how many icons we will display in the toolbar
+            for (let i = 1; i <= this.state.maxIconsDisplay; i++, idx++) {
+
+                // Make sure that we will only evaluate if we have a toolbar icon
+                if (toolbarIcons[idx]) {
+
+                    // Calculate the width of the total icons displayed
+                    totalDisplayIconWidth += toolbarIcons[idx].width;
+
+                    /*
+                     * If the width of the total icons displayed reaches the limit (toolbarContainerWidth)
+                     * Then we will break this for loop and will not add more toolbar icons to displayIcons
+                     */
+                    if (totalDisplayIconWidth > this.state.toolbarContainerWidth) {
+                        break;
+                    }
+
+                    /*
+                     * The number of icons to be displayed are limited because we need to make sure that
+                     *  they will fit in the toolbar main container's width
+                     */
+                    displayIcons.push(toolbarIcons[idx].icon);
+                }
+            }
+
+            // This will determine if we need to display the left arrow icon
+            if (this.state.startIconIndex > 0) {
+                displayArrowLeft = (
+                    <div className="chamel-toolbar-left-arrow">
+                        <FontIcon onClick={this._handleArrowClick.bind(this, -1)}
+                                  className="cfi cfi-chevron-left"/>
+                    </div>
+                );
+            }
+
+            // This will determine if we need to display the right arrow icon
+            if ((this.state.startIconIndex + displayIcons.length) < toolbarIcons.length) {
+                displayArrowRight = (
+                    <div className="chamel-toolbar-right-arrow">
+                        <FontIcon onClick={this._handleArrowClick.bind(this, 1)}
+                                  className="cfi cfi-chevron-right"/>
+                    </div>
+                );
+            }
+
+            // Modify the toolbar main container's style so we can properly display the toolbar icons
+            let styleToolbarMainCon = {
+                position: 'absolute',
+                left: DEFAULTICONWIDTH + 'px',
```

Comment:
If we only set the left (or left-margin) if we were not on page 0, then it would be better UX because we would not be wasting the valuable space for an arrow button.